bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup/Coefficient.cs

```diff
@@ -27,17 +28,81 @@ public class Coefficient : DisposableBase
     /// </summary>
     public SchnorrProof Proof { get; private set; }
 
+    /// <summary>
+    /// Generate a random coefficient
+    /// </summary>
+    public Coefficient()
+    {
+        var value = BigMath.RandQ();
+        KeyPair = ElGamalKeyPair.FromSecret(value);
+        var seed = BigMath.RandQ();
+        Proof = new(KeyPair, seed);
+    }
+
+    /// <summary>
+    /// Generate a coefficient with a predetermined value
+    /// </summary>
+    public Coefficient(ElementModQ value)
+    {
+        KeyPair = ElGamalKeyPair.FromSecret(value);
+        Proof = new SchnorrProof(KeyPair);
+    }
+
+    /// <summary>
+    /// Generate a coefficient with a predetermined value and proof seed
+    /// </summary>
+    public Coefficient(ElementModQ value, ElementModQ seed)
+    {
+        KeyPair = ElGamalKeyPair.FromSecret(value);
+        Proof = new SchnorrProof(KeyPair, seed);
+    }
+
+    /// <summary>
+    /// Generate a coefficient with a predetermined value and proof
+    /// </summary>
+    public Coefficient(ElementModQ value, SchnorrProof proof)
+    {
+        KeyPair = ElGamalKeyPair.FromSecret(value);
+
+        if (!proof.IsValid())
```

Comment:
Might want to move the IsValid check before you make the keypair.  If it is not valid do we need to make it?